temporalcli/commands.workflow.go

```diff
@@ -110,8 +125,57 @@ func (c *TemporalWorkflowQueryCommand) run(cctx *CommandContext, args []string)
 	return cctx.Printer.PrintStructured(output, printer.StructuredOptions{})
 }
 
-func (*TemporalWorkflowResetCommand) run(*CommandContext, []string) error {
-	return fmt.Errorf("TODO")
+func (c *TemporalWorkflowResetCommand) run(cctx *CommandContext, _ []string) error {
+	if c.Type.Value == "" && c.EventId <= 0 {
+		return errors.New("must specify either valid event id or reset type")
+	}
+	cl, err := c.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	resetBaseRunID := c.RunId
+	eventID := int64(c.EventId)
+	if c.Type.Value != "" {
+		resetBaseRunID, eventID, err = getResetEventIDByType(cctx, c.Type.Value, c.Parent.Namespace, c.WorkflowId, c.RunId, cl.WorkflowService())
```

Comment:
```suggestion
		resetBaseRunID, eventID, err = c.getResetEventIDByType(cctx, cl)
```

Nit, not required, just can save params and reduce the package-level pollution by using a reasonable receiver here

Reply:
Done